test/utils/createClientRender.js

```diff
@@ -114,6 +135,25 @@ const fireEvent = Object.assign(rtlFireEvent, {
     rtlFireEvent(element, event);
   },
   keyUp(element, options = {}) {
+    // `element` shouldn't be `document` but we catch this later anyway
+    const document = element.ownerDocument || element;
+    const target = document.activeElement || document.body || document.documentElement;
+    if (target !== element) {
+      // see https://www.w3.org/TR/uievents/#keyup
+      const error = new Error(
+        `\`keyup\` events can only be targeted at the active element which is ${prettyDOM(
```

Comment:
Pretty cool. Should the logic move to testing-library later on?

Reply:
Was gonna suggest/ask the same actually 🙂

Reply:
I think most people wanted to do this in user-event but also with automatically focusing. I think this is fine for app-tests but for component-tests we want to have a bit more control. Basically be as explicit as possible (fire keyDown, keyUp rather than user-event.keypress) but catch non-sensical dispatches such as fireEvent.keyDown(<section />).